velox/exec/OutputBuffer.cpp

```diff
@@ -640,4 +654,46 @@ bool OutputBuffer::isOverutilized() const {
   return (totalSize_ > maxSize_) && !atEnd_;
 }
 
+OutputBufferStats OutputBuffer::getStats() {
+  OutputBufferStats stats;
+  {
+    std::lock_guard<std::mutex> l(mutex_);
+    stats.type = PartitionedOutputNode::kindString(kind_);
+    stats.state = OutputBufferStats::BufferState::kOpen;
```

Comment:
It is not a good practice to set a variable to "wrong" value, then fix it up. Let's create a helper method to return the buffer state.

```
if (isFinishedLocked()) {
   return kFinished;
}

if (atEnd_) {
   return kFlushing;
}

...
```

Reply:
Done. Code moved to `OutputBufferStats`'s constructor with the form you propose.